File: Lv.1/최소직사각형.cpp

#include <string>
#include <vector>
#include <algorithm>
using namespace std;

int solution(vector<vector<int>> sizes) {
    int answer = 0;

    vector<int>garo;
    vector<int>sero;
    for(int i=0;i<sizes.size();i++)
    {
        if(sizes[i][1]>sizes[i][0]) //세로가 더 긴 경우 가로 세로 값을 바꿔서 garo, sero에 넣어줌
        {
            garo.push_back(sizes[i][1]);  
            sero.push_back(sizes[i][0]);
        }
        else //가로가 더 긴 경우 그대로 garo sero에 넣어줌
        {
            garo.push_back(sizes[i][0]);    
            sero.push_back(sizes[i][1]);    
        }


    }

    sort(garo.begin(),garo.end(),greater<>());
    sort(sero.begin(),sero.end(),greater<>());
    answer=garo[0]*sero[0];
    return answer;
}

File: Java/LV1/최소직사각형.java
import java.util.*;

class Solution {
    public int solution(int[][] sizes) {
        int answer = 0;
        
        int maxGaro=Math.max(sizes[0][0],sizes[0][1]);
        int maxSero=Math.min(sizes[0][0],sizes[0][1]);
        
        for(int i=0;i<sizes.length;i++){
            
            int garo = Math.max(sizes[i][0],sizes[i][1]);
            int sero = Math.min(sizes[i][0],sizes[i][1]);
            
            if(maxGaro<=garo){
                maxGaro=garo;
            }
            
            if(maxSero<=sero){
                maxSero=sero;
            }
        }
        answer=maxGaro*maxSero;
        return answer;
    }
}
